src/rgw/rgw_rest_sts.cc

```diff
@@ -289,6 +306,18 @@ int RGWSTSAssumeRole::get_params()
   serialNumber = s->info.args.get("SerialNumber");
   tokenCode = s->info.args.get("TokenCode");
 
+  if (! duration.empty()) {
+    uint64_t duration_in_secs;
+    try {
+      duration_in_secs = stoull(duration);
+    } catch (...) {
+      return -EINVAL;
+    }
+    if (duration_in_secs < STS::AssumeRoleRequest::getMinDuration() ||
```

Comment:
AssumeRoleRequestBase::validate_input() already checks whether duration is valid. The reason it is done there is because we need to get the max session duration of the role, which serves as the upper limit of the session duration here. Same for AssumeRoleWithWebIdentity()